pandas/io/formats/format.py

```diff
@@ -895,8 +895,12 @@ def _join_multiline(self, *args) -> str:
             st = ed
         return "\n\n".join(str_lst)
 
-    def to_string(self, buf: Optional[FilePathOrBuffer[str]] = None) -> Optional[str]:
-        return self.get_result(buf=buf)
+    def to_string(
+        self,
+        buf: Optional[FilePathOrBuffer[str]] = None,
+        encoding: Optional[str] = None,
+    ) -> Optional[str]:
+        return self.get_result(buf=buf, encoding=encoding)
```

Comment:
Should we be raising if both `buf` and `encoding` are `None`? I vaguely recall discussing that with @simonjayhawkins not sure where we landed

Reply:
Should I use 
```python
if (buf is None) and (encoding is None):
         raise ValueErorr('Both buf and encoding parameter are None')
or create custom exception class.
```

Reply:
> Should we be raising if both `buf` and `encoding` are `None`? I vaguely recall discussing that with @simonjayhawkins not sure where we landed

@WillAyd once #28692 is merged we can combine the tests and add some validation. see https://github.com/pandas-dev/pandas/pull/28951#pullrequestreview-301044473

Reply:
@simonjayhawkins Should that be a part of this PR itself.

Reply:
eventually.

I think best to put this on hold until #28692 is merged to ensure consistency. see https://github.com/pandas-dev/pandas/pull/28951#pullrequestreview-301044473